### PR TITLE
Initial Layout for Rocket.Chat Flutter SDK

### PR DESCRIPTION
**Pull request by yatendra2001**
## Description

This PR implements the layout and CI for Rocket.Chat Flutter SDK project as proposed in issue #1 . The project structure, CI/CD setup, and the dummy implementation of tests are included. The main packages created are `rocket_chat_embeddedchat_component` and `rocket_chat_api`.

**Packages:**

- `rocket_chat_embeddedchat_component`: Will contain the SDK with Flutter UI components for Rocket.Chat, allowing for easy embedding of the chat interface into other Flutter applications.
- `rocket_chat_api`: Provides the Dart-friendly API/stub for applications requiring in-depth modification of chat behaviour or using the chat engine as a messaging fabric.

Both packages include a sample app which will use the respective libraries to demonstrate the functionality and usage.

**CI/CD:**

- Added a GitHub Actions workflow for continuous integration, building, and testing.
- Included build scripts and tests for both packages and the sample app.
- The CI/CD will keep evolving as we move further.

**Tests:**

- Integrated dummy test directory for each package and their corresponding sample apps. This directory will contain unit_tests, integration_tests and widget_tests as we move along.

### CODE REVIEW

File: packages/rocket_chat_api/windows/flutter/generated_plugin_registrant.cc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//
+//  Generated file. Do not edit.
```
Comment (Sing-Li) [inline packages/rocket_chat_api/windows/flutter/generated_plugin_registrant.cc]: @yatendra2001   please take a look at    https://github.com/flutter/flutter/blob/master/.gitignore

These generated files should not be committed.

File: packages/rocket_chat_embeddedchat_component/sample_app/linux/flutter/generated_plugin_registrant.cc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+//
+//  Generated file. Do not edit.
```
Comment (Sing-Li) [inline packages/rocket_chat_embeddedchat_component/sample_app/linux/flutter/generated_plugin_registrant.cc]: More files that should not be committed.

File: packages/rocket_chat_embeddedchat_component/sample_app/macos/Flutter/GeneratedPluginRegistrant.swift [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//
+//  Generated file. Do not edit.
+//
```
Comment (Sing-Li) [inline packages/rocket_chat_embeddedchat_component/sample_app/macos/Flutter/GeneratedPluginRegistrant.swift]: More files to ignore.

**Review by Sing-Li — CHANGES_REQUESTED** [review]
Please fix some of the minor `.gitignore` problems.  Thanks.

**Comment by Sing-Li** [comment]
Need also to make sure the workflows are legit by adding one of the monorepo management tools.   

Seems `Melos` ,  `mono_repo`,  `flutter_monorepo_build_tools` or other.

**Comment by yatendra2001** [comment]
@Sing-Li I've added new .gitignore rules and added Melos as a monorepo management tool. Also, should we make some changes in Readme.md file too?

**Review by Sing-Li — APPROVED** [review]
looks like good enough for the first commit.  Thx

**Comment by Sing-Li** [comment]
closes #1

**Pull request merged**